File: Frontend/Certiva/src/app/Models/dashboard-dto.ts
export interface DashboardDTO {
  totalUsuarios: number;
  eventosActivos: number;
  asistenciasTotales: number;
  certificadosEmitidos: number;
  distribucionRoles: Record<string, number>;
}

export interface DashboardActivityPointDTO {
  fecha: string;
  asistencias: number;
  certificados: number;
}

export interface DashboardActivityDTO {
  rangoDias: number;
  puntos: DashboardActivityPointDTO[];
}

File: Backend/api/src/main/java/com/certiva/api/DTO/DashboardDTO.java
package com.certiva.api.DTO;

import java.util.HashMap;
import java.util.Map;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DashboardDTO {

    private Long totalUsuarios;

    private Long eventosActivos;

    private Long asistenciasTotales;

    private Long certificadosEmitidos;

    @Builder.Default
    private Map<String, Long> distribucionRoles = new HashMap<>();
}
